### ISSUE TITLE
Documentation on migrations is lacking

### ISSUE THREAD

**Issue by ffxsam** [tgriesser/knex#3129]
IMO the documentation on migrations is pretty thin, and leaves a lot of questions.

For instance, this is shown in the docs:

```js
exports.up = function(knex, Promise) { /* ... */ };

exports.down = function(knex, Promise) { /* ... */ };
```

But it doesn't give any explanation of how to actually write migrations. What is `Promise` for? The migrations section starts of well enough, explaining the CLI commands. But then:

>Once you have finished writing the migrations, you can update the database matching your NODE_ENV by running:

and you're left to guess how this is done.

**Comment by zachelrath** [comment]
@ffxsam Agreed, these could be fleshed out a bit more with a basic example. I also think that `Promise` could be removed from the docs, as it's not a commonly-used parameter in my experience. It's easy enough to use either a native Promise or to import a more full-featured Promise library like Bluebird if you really need some fancy Promise semantics.

What do you think about this example?

```
exports.up = function(knex) {
   return knex.schema
      .createTable('user', function (table) {
         table.increments('id');
         table.string('first_name', 255).notNullable();
         table.string('last_name', 255).notNullable();
      })
      .createTable('product', function (table) {
         table.increments('id');
         table.decimal('price').notNullable();
         table.string('name', 1000).notNullable();
      });
};

exports.down = function(knex) {
    return knex.schema
        .dropTable("product")
        .dropTable("user");
};
```

**Comment by ffxsam** [comment]
@zachelrath I think that example is perfect. It provides enough of a starting point for a dev to go "Oh ok, I need to look at the schema builder section of the docs." As it stands now, it just kind of leaves you hanging with no idea what to do.

**Comment by zachelrath** [comment]
@ffxsam I submitted a PR to the docs repo for this change: https://github.com/knex/documentation/pull/187

**Comment by elhigu** [comment]
fixed by knex/documentation#187
